File: backend/src/main/java/cn/etarch/mao/session/activity/ActivityTypeMapper.java
package cn.etarch.mao.session.activity;

public class ActivityTypeMapper {

    public static String mapToolToType(String toolName) {
        if (toolName == null) return "TOOL";
        return switch (toolName.toLowerCase()) {
            case "read_file" -> "READ";
            case "write_file", "edit_file" -> "EDIT";
            case "shell" -> "RUN";
            case "glob", "list" -> "EXPLORE";
            case "task_create", "task_update", "task_delete", "task_list" -> "TASK";

            default -> "TOOL";
        };
    }
}

File: backend-ts/src/session/activity/activity-type-mapper.ts
export function mapToolToType(toolName: string | null | undefined): string {
  if (toolName == null) {
    return 'TOOL';
  }
  switch (toolName.toLowerCase()) {
    case 'read_file':
      return 'READ';
    case 'write_file':
    case 'edit_file':
      return 'EDIT';
    case 'shell':
      return 'RUN';
    case 'glob':
    case 'list':
      return 'EXPLORE';
    case 'task_create':
    case 'task_update':
    case 'task_delete':
    case 'task_list':
      return 'TASK';
    default:
      return 'TOOL';
  }
}

export const ActivityTypeMapper = { mapToolToType };
